sdk/typescript/src/eth-light-client/eth-light-client.ts

```diff
@@ -0,0 +1,305 @@
+	let contractAddressBcs = bcs.vector(bcs.u8()).serialize(contractAddressArrayU8);
+
+	const tx = new TransactionBlock();
+	tx.moveCall({
+		target: `${packageId}::${ethereumStateModuleName}::init_state`,
+		arguments: [
+			tx.pure(stateBcs),
+			tx.pure(network),
+			tx.pure(contractAddressBcs),
+			tx.pure.u64(contractApprovedTxSlot),
+			tx.pure(updatesBcs),
+			tx.pure(finalityUpdateBcs),
+			tx.pure(optimisticUpdateBcs),
+		],
+		typeArguments: [],
+	});
+	tx.setGasBudget(20000000000);
+
+	let result = await client.signAndExecuteTransactionBlock({
+		signer: keypair,
+		transactionBlock: tx,
+		options: { showEffects: true },
+	});
+
+	return result.effects?.created?.filter(
+		(o) =>
+			typeof o.owner === 'object' &&
+			'Shared' in o.owner &&
+			o.owner.Shared.initial_shared_version !== undefined,
+	)[0].reference!.objectId!;
+};
+
+/**
+ * Approves an Ethereum transaction for a given dWallet.
+ *
+ * Interacts with the Ethereum light client to verify and approve a transaction message
+ * using an Ethereum smart contract linked to a dWallet within the dWallet blockchain context.
+ * The verification of the state and message is done offline, inside the dWallet module.
+ *
+ * **Logic**
+ * 1. **Retrieve Configuration**: Starts by retrieving the latest Ethereum state object.
+ * 2. **Fetch Ethereum Objects**: Retrieves and deserializes the latest Ethereum state and the current Ethereum state data to collect the latest Ethereum state data.
+ * 3. **Initialize Light Client**: Initializes the Ethereum light client with the deserialized Ethereum state.
+ * 4. **Prepare Proof Parameters**: Constructs proof request parameters using the message, dWallet ID, and data slot from the latest Ethereum state object.
+ * 5. **Fetch Updates and Proofs**: Retrieves the necessary updates and cryptographic proofs from the Ethereum light client.
+ * 6. **Build Transaction**: Uses the transaction builder to serialize transaction parameters, including the Ethereum state, updates, and shared state object, and prepares the transaction to call the `verify_new_state` function in the Ethereum state module.
+ * 7. **Send Transaction**: Constructs the transaction data, including the proof and dWallet ID, and executes it.
+ *
+ * **Arguments**
+ * @param {string} ethDwalletCapId - The ObjectID of the Ethereum dWallet capability, representing the link between the dWallet and Ethereum.
+ * @param {string} message - The Ethereum transaction message to be approved.
+ * @param {string} dWalletID - The ObjectID of the dWallet to which the transaction belongs.
+ * @param {string} latestStateObjectID - The ObjectID of the latest Ethereum state.
+ * @param {string} executionRpc - The Ethereum execution RPC endpoint.
+ * @param {string} consensusRpc - The Ethereum consensus RPC endpoint.
+ * @param {Keypair} keypair - The keypair used to sign the transaction.
+ * @param {DWalletClient} client - The dWallet client instance.
+ */
+export const approveEthereumMessage = async (
```

Comment:
This function is 100 LOC - maybe break it to two internal functions - one that updates the light client state and one that approves the message?